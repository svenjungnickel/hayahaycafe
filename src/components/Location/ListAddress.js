import React from 'react';
import PropTypes from 'prop-types';
import AddressItem from './AddressItem';
import ListAddressStyles from '../../styles/components/Location/ListAddress.module.scss';

const ListAddress = ({ address }) => {
    const filteredAddressItems = Object.values(address).filter((item) => '' !== item);

    return (
        <div className={ListAddressStyles.listAddress}>
            <h2>Location</h2>
            {0 < filteredAddressItems.length &&
                filteredAddressItems.map((item, key) => {
                    return <AddressItem item={item} key={key} />;
                })}
        </div>
    );
};

ListAddress.propTypes = {
    address: PropTypes.shape({
        company: PropTypes.string,
        addressLine1: PropTypes.string,
        addressLine2: PropTypes.string,
        province: PropTypes.string,
        postalCode: PropTypes.string,
        country: PropTypes.string,
    }).isRequired,
};

export default ListAddress;
