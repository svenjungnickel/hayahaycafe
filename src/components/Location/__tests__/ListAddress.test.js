import React from 'react';
import renderer from 'react-test-renderer';
import ListAddress from '../ListAddress';

describe('Listaddress', () => {
    it('renders without address throws prop type error', () => {
        const renderListAddress = () => {
            const component = <ListAddress />;
            renderer.create(component);
        };

        expect(renderListAddress).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid address data throws prop type error', () => {
        const renderListAddress = () => {
            const address = {
                company: '',
                addressLine1: 'addressLine1',
                addressLine2: '',
                province: 'province',
                postalCode: 12345,
                country: '',
            };

            const component = <ListAddress address={address} />;
            renderer.create(component);
        };

        expect(renderListAddress).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly ', () => {
        const address = {
            company: '',
            addressLine1: 'addressLine1',
            addressLine2: '',
            province: 'province',
            postalCode: '12345',
            country: '',
        };

        const component = <ListAddress address={address} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
