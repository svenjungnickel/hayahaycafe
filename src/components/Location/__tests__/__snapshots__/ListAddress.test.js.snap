// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Listaddress renders correctly 1`] = `
<div
  className="listAddress"
>
  <h2>
    Location
  </h2>
  addressLine1
  <br />
  province
  <br />
  12345
  <br />
</div>
`;
